backend/src/openbeheer/api/views.py

```diff
@@ -253,3 +263,80 @@ def paginate(
             ),
             results=data.results,
         )
+
+
+class DetailView[T](MsgspecAPIView, ABC):
+    data_type: type[T]
+    has_versions: bool
+    endpoint_path: str
+
+    def get_item_data(self, slug: str, uuid: str) -> tuple[T | ValidatieFout, int]:
+        with ztc_client() as client:
+            response = client.get(
+                self.endpoint_path.format(uuid=uuid),
+            )
+
+        if not response.ok:
+            # error = decode(response.content, type=ValidatieFout) # TODO: the OZ 404 response gives invalid JSON back
+            return ValidatieFout(
+                code="",
+                title="",
+                detail="",
+                instance="",
+                status=response.status_code,
+                invalid_params=[],
+            ), response.status_code
+
+        content = response.content
+        try:
+            return decode(
+                content,
+                type=self.data_type,
+                strict=False,
+            ), response.status_code
+        except ValidationError as e:
+            return ValidatieFout(
+                code="Bad response",
+                title="Server returned out of spec response",
+                detail=str(e),
+                instance="",
+                status=500,
+                invalid_params=[],
+            ), 500
+
+    def get(self, request: Request, slug: str, uuid: str, *args, **kwargs) -> Response:
```

Comment:
```suggestion
    def get(self, request: Request, slug: str, uuid: UUID, *args, **kwargs) -> Response:
```

? Or doesn't `<uuid:uuid>` in the path actually convert to UUID?